packages/liveblocks-core/src/room.ts

```diff
@@ -893,8 +894,40 @@ export function createRoom<
       ? options.initialStorage(config.roomId)
       : options.initialStorage;
 
+  const doc = typeof document !== "undefined" ? document : undefined;
+  let inBackgroundSince: number | null = null;
+
+  function onVisibilityChange() {
+    if (doc?.visibilityState === "hidden") {
```

Comment:
nit, but maybe consider early out unless config.backgroundKeepAliveTimeout is defined.

Reply:
I wanted to keep this logic generic and since it's just keeping track of a number, I don't think it's worth polluting it here. In a75a08ab8c0bb96c1eeae06990efed9f08d85af0, I've refactored this thing out into a helper function that's independent of the room at all, so we don't have to pollute the function with unnecessary local variables/functions that only complicate the already-large `createRoom` function.